documentation/writing_tests.md

```diff
@@ -0,0 +1,1534 @@
+
+Chai makes testing much easier by providing an extensive list of assertions that can run against your code. This list of assertions can be seen in detail by going to their assertion style guide: <https://www.chaijs.com/guide/styles/>
+
+Another important learning resource for Chai is: Chai Assertions for Promises <https://www.chaijs.com/plugins/chai-as-promised/>.
+
+#### Recommended Chai assertion style
+
+The Azure SDK for JavaScript (and TypeScript) prefers Chai's Assert style. For example:
+
+```ts
+import { assert } from "chai";
+
+let foo = "bar";
+let beverages = { tea: [ 'chai', 'matcha', 'oolong' ] };
+
+assert.typeOf(foo, "string");
+assert.typeOf(foo, "string", "foo is a string");
+assert.equal(foo, "bar", "foo equals to `bar`");
+assert.lengthOf(foo, 3, 'foo`s value has a length of 3');
+assert.lengthOf(beverages.tea, 3, 'beverages has 3 types of tea');
+```
+
+#### Chai in our dependencies
+
+Since we're using [Rush](https://rushjs.io/), we have to use the same version of our packages in each one of the projects inside of this repository. If you want to add `chai` as a dev dependency to a new project inside of this repository, first make sure you are in the root folder of that project, then you can run the following command:
+
+```
+rush add --dev -p chai
+```
+
+Since we're using TypeScript, make sure to also add the types:
+
+```
+rush add --dev -p @types/chai
+```
+
+### Rollup
+
+Rollup is a module bundler for JavaScript. It uses the new standardized format for code modules included in the ES6 revision of JavaScript. This will eventually be possible natively everywhere, but Rollup lets you do it today.
+
+We use Rollup to take the result of the TypeScript compiler and carefully bundle it differently depending on the target platform (NodeJS or the browsers) with:
+
+- The source maps available from our compiled TypeScript and our dependencies.
+- Without specific sections of code that exist only for a specific platform.
+- To transform the necessary dependencies from our `node_modules`, from CommonJS to ES6. 
+- With a banner including Microsoft's copyright at the top of the generated file.
+- With the consideration necessary to output a working JavaScript file compatible with browsers.
+- With special settings necessary to run our tests in the browsers.
+
+If you want to learn more about Rollup, you can read the Rollup guide at: https://rollupjs.org/guide/en/
+
+#### Rollup in our dependencies
+
+Since we're using [Rush](https://rushjs.io/), we're forced to use the same version of our packages in each one of the projects inside of this repository. If you want to add `chai` as a dev dependency to a new project, once you're at the root of this new project, you can run the following command:
```

Comment:
Changing this to:

> If you want to add `chai` as a dev dependency to a new project, please use [Rush](https://rushjs.io/) since we're most likely using the same version of Chai in more projects. Once you're at the root of this new project, you can run the following command: